server/src/main/java/org/apache/druid/server/coordinator/compact/NewestSegmentFirstIterator.java

```diff
@@ -110,9 +110,10 @@ public class NewestSegmentFirstIterator implements CompactionSegmentIterator
             // For example, if the original is interval of 2020-01-28/2020-02-03 with WEEK granularity
             // and the configuredSegmentGranularity is MONTH, the segment will be split to two segments
             // of 2020-01/2020-02 and 2020-02/2020-03.
-            if (Intervals.ETERNITY.equals(segment.getInterval())) {
+            if (Intervals.ETERNITY.getStart().equals(segment.getInterval().getStart())
+                || Intervals.ETERNITY.getEnd().equals(segment.getInterval().getEnd())) {
               // This is to prevent the coordinator from crashing as raised in https://github.com/apache/druid/issues/13208
-              log.warn("Cannot compact datasource[%s] with ALL granularity", dataSource);
+              log.warn("Cannot compact datasource[%s] containing segments with partial-ETERNITY intervals", dataSource);
```

Comment:
While at it, could you add the segment id as well? It'd make it easy to take any action in the metadata store or via the console, if needed.

Reply:
There could be several segments and it could be noisy to log all of them